huaweicloud/services/dataarts/resource_huaweicloud_dataarts_architecture_code_table.go

```diff
@@ -0,0 +1,753 @@
+			return diag.Errorf("error adding field values to DataArts Architecture code table: %s", err)
+		}
+	}
+
+	return resourceArchitectureCodeTableRead(ctx, d, meta)
+}
+
+func buildCreateCodeTableParams(d *schema.ResourceData) map[string]interface{} {
+	return map[string]interface{}{
+		"name_ch":           d.Get("name"),
+		"name_en":           d.Get("code"),
+		"directory_id":      d.Get("directory_id"),
+		"code_table_fields": buildCreateCodeTableFieldParams(d),
+		"description":       utils.ValueIngoreEmpty(d.Get("description")),
+	}
+}
+
+func buildCreateCodeTableFieldParams(d *schema.ResourceData) []map[string]interface{} {
+	rawFields := d.Get("fields").([]interface{})
+	fields := make([]map[string]interface{}, len(rawFields))
+	for i, rawField := range rawFields {
+		fieldMap := rawField.(map[string]interface{})
+		field := make(map[string]interface{})
+		field["ordinal"] = i + 1
+		field["name_ch"] = fieldMap["name"].(string)
+		field["name_en"] = fieldMap["code"].(string)
+		field["data_type"] = fieldMap["type"].(string)
+		field["description"] = utils.ValueIngoreEmpty(fieldMap["description"].(string))
+		fields[i] = field
+	}
+	return fields
+}
+
+func insertCodeTableFieldValues(client *golangsdk.ServiceClient, d *schema.ResourceData) error {
+	ids, err := getCodeTableFieldID(client, d)
+	if err != nil {
+		return err
+	}
+	var insertCodeTableFieldValuesHttpUrl = "v2/{project_id}/design/code-tables/{id}/values"
+	insertCodeTableFieldValuesPath := client.Endpoint + insertCodeTableFieldValuesHttpUrl
+	insertCodeTableFieldValuesPath = strings.ReplaceAll(insertCodeTableFieldValuesPath, "{project_id}", client.ProjectID)
+	insertCodeTableFieldValuesPath = strings.ReplaceAll(insertCodeTableFieldValuesPath, "{id}", d.Id())
+
+	insertCodeTableFieldValuesOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		MoreHeaders:      map[string]string{"workspace": d.Get("workspace_id").(string)},
+	}
+
+	insertCodeTableFieldValuesOpt.JSONBody = utils.RemoveNil(buildInsertCodeTableFieldValuesParams(d, ids))
+	_, err = client.Request("PUT", insertCodeTableFieldValuesPath, &insertCodeTableFieldValuesOpt)
+
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func getCodeTableFieldID(client *golangsdk.ServiceClient, d *schema.ResourceData) ([]interface{}, error) {
+	getCodeTableResp, err := readCodeTable(client, d)
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving DataArts Architecture code table: %s", err)
+	}
+
+	createCodeTableRespBody, err := utils.FlattenResponse(getCodeTableResp)
+	if err != nil {
+		return nil, err
+	}
+
+	codeTable := utils.PathSearch("data.value", createCodeTableRespBody, nil)
+
+	fieldsIDs, err := jmespath.Search("code_table_fields[*].id", codeTable)
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving DataArts Architecture code table fields: %s is not found in API response", "id")
+	}
+
+	return fieldsIDs.([]interface{}), nil
+}
+
+func hasFieldValues(d *schema.ResourceData) bool {
+	count := false
```

Comment:
`has` or `ok` will make more sense.
```suggestion
	has := false
```